DB/src/main/java/io/deephaven/db/v2/parquet/ParquetReaderUtil.java

```diff
@@ -101,6 +104,285 @@ public static void readParquetFile(String filePath) throws IOException {
+                return Optional.empty();
+            }
+
+            @Override
+            public Optional<Class<?>> visit(final LogicalTypeAnnotation.IntervalLogicalTypeAnnotation intervalLogicalType) {
+                errorString.setValue("IntervalLogicalType");
+                return Optional.empty();
+            }
+
+            @Override
+            public Optional<Class<?>> visit(final LogicalTypeAnnotation.MapKeyValueTypeAnnotation mapKeyValueLogicalType) {
+                errorString.setValue("MapLogicalType");
+                return Optional.empty();
+            }
+        };
+
+        final String csvGroupingCols = keyValueMetaData.get(ParquetTableWriter.GROUPING);
+        Set<String> groupingCols = Collections.emptySet();
+        if (csvGroupingCols != null) {
+            groupingCols = new HashSet<>(Arrays.asList(csvGroupingCols.split(",")));
+        }
+
+        int c = 0;
+        for (ColumnDescriptor column : schema.getColumns()) {
+            currentColumn.setValue(column);
+            final LogicalTypeAnnotation logicalTypeAnnotation = column.getPrimitiveType().getLogicalTypeAnnotation();
+            final String colName = schema.getFieldName(c++);
+            final boolean isGroupinng = groupingCols.contains(colName);
+            final String codecName = keyValueMetaData.get(ParquetTableWriter._CODEC_NAME_PREFIX_ + colName);
+            final String codecArgs = keyValueMetaData.get(ParquetTableWriter._CODEC_ARGS_PREFIX_ + colName);
+            String codecType = keyValueMetaData.get(ParquetTableWriter._CODEC_TYPE_PREFIX_ + colName);
+            if (codecType != null && !codecType.isEmpty()) {
+                final Class<?> dbType = loadClass(colName, "codec type", codecType);
+                final String codecComponentType = keyValueMetaData.get(ParquetTableWriter._CODEC_COMPONENT_TYPE_PREFIX_ + colName);
+                final Class<?> componentType;
+                if (codecComponentType == null || codecComponentType.isEmpty()) {
+                    componentType = null;
+                } else {
+                    final Class<?> maybePrimitiveType = classForPrimitiveNameOrNull(codecComponentType);
+                    componentType = (maybePrimitiveType != null)
+                        ? maybePrimitiveType
+                        : loadClass(colName, "codec component type", codecComponentType)
+                        ;
+                }
+                colDefConsumer.accept(colName, dbType, componentType, isGroupinng, codecName, codecArgs);
+                continue;
+            }
+            final boolean isArray = column.getMaxRepetitionLevel() > 0;
+            if (logicalTypeAnnotation == null) {
+                switch (column.getPrimitiveType().getPrimitiveTypeName()) {
+                    case BOOLEAN:
+                        if (isArray) {
+                            colDefConsumer.accept(colName, Boolean[].class, Boolean.class, isGroupinng, codecName, codecArgs);
```

Comment:
You're missing a db-array choice here.

Reply:
Not for this diff; as it exists today, writing of DbArrays uses binary encoding with a codec; we discussed we want to change this in the future so that loading DH-generated parquet files in other tools is more transparent (so we will store the DbArrays as an array of the type in parquet, plus a new metadata annotation, but that change has not happened yet; it implies making special_type support dbarrays etc)

Reply:
In the future we want to write DbArrays as parquet native repeated types (their "natural" representation).  But in the rest of the parquet code as it predates this diff (remember what this ticket's title is), DbArrays are written as binary and being encoded.

Reply:
Is that statement true? I had the impression from the ReadOnlyParquetTableLocation or whatever that we might have DbArrays and Arrays stored the same way, and used dataType from the ColumnDefinition to disambiguate.

Reply:
I was wrong, we are storing DbArrays of a primitive composite type as repeated of the primitive, and this works both ways.  The issue we saw in the testcase I added to TableManagementTools (in issue https://github.com/deephaven/deephaven-core/issues/726) is for DbArrays of DbArrays (component type DbArrayDouble actually).